### PR TITLE
Local development and guide

### PR DESCRIPTION
**Pull request by mhmxs**
## Description

This changes is about how to develop and test Trousseau locally. With some fixes found during testing. (see individual comments)

Please include: 
- [X] a summary of the change and which issue is fixed
- [X] relevant motivation and context
- [X] any dependencies that are required for this change

Please check options that are relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Manually

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: Taskfile.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ vars:
   HELM_VERSION: 3.6.3
   VAULT_VERSION: 1.8.1
   KUBECTL_VERSION: 1.21.1
-  SCRIPT: scripts/templates/testing
+  SCRIPT: scripts/hcvault/archives/testing
```
Comment (mhmxs) [inline Taskfile.yml:10]: Directory not found, i have found the content in the new place.

File: Taskfile.yml [hunk 2]
```diff
@@ -126,11 +126,10 @@ tasks:
   docker:build:
     desc: build docker image
     cmds:
-      - docker build -t $DOCKER_REGISTRY/$IMAGE_NAME:latest .
+      - docker build -t $DOCKER_REGISTRY/$IMAGE_NAME:$IMAGE_VERSION .
```
Comment (mhmxs) [inline Taskfile.yml:129]: Kubernetes always fetches `latest` so this tag is not ok for local development and testing at all.

File: scripts/hcvault/archives/testing/kind-cluster.yaml [hunk 3]
```diff
@@ -5,7 +5,7 @@ nodes:
 - role: control-plane
   extraMounts:
   - containerPath: /etc/kubernetes/encryption-config.yaml
-    hostPath: scripts/encryption-config.yaml
+    hostPath: scripts/hcvault/archives/testing/encryption-config.yaml
```
Comment (mhmxs) [inline scripts/hcvault/archives/testing/kind-cluster.yaml:8]: Directory not found, i have found the content in the new place.

File: Taskfile.yml (hunk 2 above)
Comment (rovandep) [reply Taskfile.yml:129]: agreed!

**Review by rovandep — APPROVED** [review]

**Pull request merged**